src/components/NFTPage/Recommendation.vue

```diff
@@ -101,11 +103,17 @@ export default {
     },
   },
   methods: {
+    handleHeaderAvatarClick() {
+      this.$emit('header-avatar-click');
+    },
     handleFollowButtonClick() {
-      this.$emit('on-follow-button-click');
+      this.$emit('follow-button-click');
+    },
+    handleItemClick(classId) {
+      this.$emit('item-click', classId);
```

Comment:
I don't know why this emit cannot be received by the parent, still investigating...

Reply:
Is th function actually triggered? If not maybe we should manually emit some event in nftPortfolioItem's nuxtLink's click.native

Reply:
Shouldn't trigger event in `<NFTPortfolioItem/>` as we need to emit different events for`<NFTPortfolioPage/>`, `<NFTDetailsPage/>` and `<NFTClassDetailsPage/>`.

I have made a temp fix, we can investigate this later so we can start receiving events asap